internal/servers/controller/handlers/errors.go

```diff
@@ -175,14 +174,16 @@ func ErrorHandler(logger hclog.Logger) runtime.ErrorHandlerFunc {
 			}
 		}
 
-		if apiErr == nil || apiErr.inner.GetStatus() == http.StatusInternalServerError {
-			errId, err := internalErrorId()
-			if err != nil {
-				logger.Error("unable to generate internal error id", "error", err)
-				errId = "failed_to_generate_error_id"
-			}
-			logger.Error("internal error returned", "error id", errId, "error", inErr)
-			apiErr = getInternalError(errId)
+		traceId := r.Header.Get("TraceId")
+		if traceId == "" {
+			traceId = "failed_to_get_trace_id"
```

Comment:
If there's no trace ID, this should be blank, not a non-valid trace ID value

Reply:
I removed the trace id generation and instead we return the full error message with 500 responses.